apps/src/templates/instructions/InstructionsWithWorkspace.jsx

```diff
@@ -0,0 +1,183 @@
+var connect = require('react-redux').connect;
+var utils = require('../../utils');
+var styleConstants = require('../../styleConstants');
+var CodeWorkspaceContainer = require('../CodeWorkspaceContainer');
+var TopInstructions = require('./TopInstructions');
+var instructions = require('../../redux/instructions');
+
+var HEADER_HEIGHT = styleConstants['workspace-headers-height'];
+var RESIZER_HEIGHT = styleConstants['resize-bar-width'];
+
+/**
+ * A component representing the right side of the screen in our app. In particular
+ * it has instructions in the top pane (unless disabled), a resizer, and then
+ * our code workspace component
+ */
+var InstructionsWithWorkspace = React.createClass({
+  propTypes: {
+    showInstructions: React.PropTypes.bool.isRequired,
+    instructionsCollapsed: React.PropTypes.bool.isRequired,
+    instructionsHeight: React.PropTypes.number.isRequired,
+    instructionsMaxHeight: React.PropTypes.number.isRequired,
+
+    generateCodeWorkspaceHtml: React.PropTypes.func.isRequired
```

Comment:
I assume the extra newline here separates props provided by Redux from props that need to be provided externally.  Might be nice to call that out explicitly with comments.